src/layouts/Workspace/WorkspacePagesAndFoldersComponents.tsx

```diff
@@ -0,0 +1,233 @@
+import {
+  SimpleGrid,
+  Box,
+  Skeleton,
+  Text,
+  Button,
+  Center,
+  Icon,
+  VStack,
+  IconButton,
+  ButtonGroup,
+} from "@chakra-ui/react"
+import { ContextMenu } from "components/ContextMenu"
+import {
+  BiBulb,
+  BiChevronDown,
+  BiChevronUp,
+  BiFileBlank,
+  BiFolder,
+  BiInfoCircle,
+} from "react-icons/bi"
+import { Link } from "react-router-dom"
+
+import useRedirectHook from "hooks/useRedirectHook"
+
+import { EmptyBoxImage } from "assets/images/EmptyBoxImage"
+import { DirectoryData, PageData } from "types/directory"
+
+import {
+  Section,
+  SectionHeader,
+  SectionCaption,
+  CreateButton,
+} from "../components"
+
+import { PageCard, FolderCard } from "./components"
+
+export const Folders = (props: {
+  siteName: string
+  pagesData: PageData[]
+  url: string
+  dirsData: (PageData | DirectoryData)[]
```

Comment:
hmm this is getting abit out of hand - we should probably create a type alias for readability.

also why is `dirsData` typed as `(PageData | DirectoryData)[]`? wouldn't `DirectoryData[]` suffice?